### PR TITLE
Convert some code under containers/Auth to TS, refactor to not use deprecated utils

### PR DESCRIPTION
**Pull request by nl0**

### CODE REVIEW

File: catalog/app/containers/Auth/Layout.tsx [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+export function Heading(props: M.TypographyProps) {
+  return <M.Typography variant="h4" align="center" {...props} />
+}
+
+interface FieldOwnProps {
+  // TODO: use redux-form / final-form type definitions
+  input: {}
+  meta: {
+    error: any
+    submitError: any
```
Comment (fiskus) [inline catalog/app/containers/Auth/Layout.tsx:30]: It seems that code expects `error` and `submitError` to be string:

```
    error: string | null
    submitError: string | null
```
or
```
    error?: string
    submitError?: string
```

**Review by fiskus — APPROVED** [review]

File: catalog/app/containers/Auth/Layout.tsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+const useContainerStyles = M.makeStyles({
+  root: {
+    marginLeft: 'auto',
+    marginRight: 'auto',
+    maxWidth: 300,
+    width: '100%',
+  },
+})
+
+export function Container(props: JSX.IntrinsicElements['div']) {
```
[the remark this answers is not in the document]
Comment (nl0) [reply catalog/app/containers/Auth/Layout.tsx:16]: yep, makes sense, i'll adjust that

File: catalog/app/containers/Auth/Layout.tsx (hunk 1 above)
Comment (nl0) [reply catalog/app/containers/Auth/Layout.tsx:30]: yep, it seems that way in this case, but generally redux-form and final form accepts anything as a validation error

**Comment by nl0** [comment]
@fiskus thx for the feedback, i've applied your suggestions and rebased the PR, gonna merge now

**Pull request merged**